src/web/TextInput.tsx

```diff
@@ -122,16 +125,29 @@ export class TextInput extends React.Component<Types.TextInputProps, TextInputSt
                     onMouseUp={ this._checkSelectionChanged }
                     onPaste={ this._onPaste }
                     aria-label={ this.props.accessibilityLabel }
-                    type={ this.props.secureTextEntry ? 'password' : this._getKeyboardType() }
+                    type={ this.props.secureTextEntry ? 'password' : keyboardType }
```

Comment:
You should fold the check for ```this.props.secureTextEntry``` into the ```_getKeyboardType``` method so the logic is all in one place.